src/Simulation/EntryPointDriver/Driver.cs

```diff
@@ -241,146 +291,220 @@ private void AddOptionIfAvailable<T>(Command command, OptionInfo<T> option)
+            IEnumerable<(Command, IEnumerable<ValidateSymbol<CommandResult>>)> entryPointCommands)
+        {
+            var validators = Enumerable.Empty<ValidateSymbol<CommandResult>>();
 
-        /// <summary>
-        /// The resource group option.
-        /// </summary>
-        internal static readonly OptionInfo<string> ResourceGroupOption = new OptionInfo<string>(
-            ImmutableList.Create("--resource-group"), "The resource group name.");
+            var simulate = new Command("simulate", "(default) Run the program using a local simulator.");
+            if (entryPointCommands.Count() == 1)
+            {
+                var (epCommand, valids) = entryPointCommands.First();
+                simulate.AddCommand(epCommand);
+                SetSubCommandAsDefault(simulate, epCommand, valids);
+                validators = valids;
+            }
+            else
+            {
+                foreach (var (epCommand, _) in entryPointCommands)
+                {
+                    simulate.AddCommand(epCommand);
+                }
+            }
 
-        /// <summary>
-        /// The workspace option.
-        /// </summary>
-        internal static readonly OptionInfo<string> WorkspaceOption = new OptionInfo<string>(
-            ImmutableList.Create("--workspace"), "The workspace name.");
+            return (simulate, validators);
+        }
 
         /// <summary>
-        /// The storage option.
+        /// Creates the Azure submit command.
         /// </summary>
-        internal static readonly OptionInfo<string?> StorageOption = new OptionInfo<string?>(
-            ImmutableList.Create("--storage"), default, "The storage account connection string.");
+        /// <param name="entryPointCommands">The entry point commands that will be the sub commands to the created command.</param>
+        /// <returns>Tuple with the created submit command and the validators for that command.</returns>
+        private static (Command, IEnumerable<ValidateSymbol<CommandResult>>) CreateSubmitCommand(IEnumerable<(Command, IEnumerable<ValidateSymbol<CommandResult>>)> entryPointCommands)
+        {
+            var validators = Enumerable.Empty<ValidateSymbol<CommandResult>>();
 
-        /// <summary>
-        /// The AAD token option.
-        /// </summary>
-        internal static readonly OptionInfo<string?> AadTokenOption = new OptionInfo<string?>(
-            ImmutableList.Create("--aad-token"), default, "The Azure Active Directory authentication token.");
+            var submit = new Command("submit", "Submit the program to Azure Quantum.")
+            {
+                IsHidden = true,
+            };
+            if (entryPointCommands.Count() == 1)
+            {
+                var (epCommand, valids) = entryPointCommands.First();
+                submit.AddCommand(epCommand);
+                SetSubCommandAsDefault(submit, epCommand, valids);
+                validators = valids;
```

Comment:
nit: Same as above - you can return from both branches instead of using a mutable variable.

Reply:
done